src/topology/homotopy/H_spaces.lean

```diff
@@ -0,0 +1,233 @@
+
+lemma continuous_Q_right : continuous Q_right :=
+continuous_proj_Icc.comp $ continuous.div (by continuity) (by continuity) (λ x, one_add_pos.ne')
+
+lemma Q_right_zero_left (θ : I) : Q_right (0, θ) = 0 :=
+set.proj_Icc_of_le_left _ $ by simp only [coe_zero, mul_zero, zero_div]
+
+lemma Q_right_one_left (θ : I) : Q_right (1, θ) = 1 :=
+set.proj_Icc_of_right_le _ $ (le_div_iff one_add_pos).2 $
+  by { dsimp only, rw [coe_one, one_mul, mul_one], apply add_le_add_left (le_one _) }
+
+lemma Q_right_zero_right (t : I) : (Q_right (t, 0) : ℝ) = if (t : ℝ) ≤ 1 / 2 then 2 * t else 1 :=
+begin
+  simp only [Q_right, coe_zero, add_zero, div_one],
+  split_ifs,
+  { rw set.proj_Icc_of_mem _ ((mul_pos_mem_iff zero_lt_two).2 _), exacts [rfl, ⟨t.2.1, h⟩] },
+  { rw (set.proj_Icc_eq_right _).2, { refl }, { linarith }, { exact zero_lt_one } },
+end
+
+lemma Q_right_one_right (t : I) : Q_right (t, 1) = t :=
+eq.trans (by {rw Q_right, congr, apply mul_div_cancel_left, exact two_ne_zero}) $
+  set.proj_Icc_coe zero_le_one _
+
+variables {x y : X}
+
+/- This is the function analogous to the one on p. 475 of Serre's *Homologie singulière des espaces
+  fibrés*, defining a homotopy from the product path `e ∧ γ` to `γ`.-/
+def delayed_refl_right (θ : I) (γ : path x y) : path x y :=
+{ to_fun := λ t, γ (Q_right (t, θ)),
+  continuous_to_fun := γ.continuous.comp (continuous_Q_right.comp $ continuous.prod.mk_left θ),
+  source' := by { dsimp only, rw [Q_right_zero_left, γ.source] },
+  target' := by { dsimp only, rw [Q_right_one_left, γ.target] } }
+
+lemma continuous_delayed_refl_right {x : X} :
+  continuous (λ p : I × path x y, delayed_refl_right p.1 p.2) :=
+continuous_uncurry_iff.mp $ (continuous_snd.comp continuous_fst).path_eval $
+  continuous_Q_right.comp $ continuous_snd.prod_mk $ continuous_fst.comp continuous_fst
+
+lemma delayed_refl_right_zero (γ : path x y) : delayed_refl_right 0 γ = γ.trans (path.refl y) :=
+begin
+  ext t,
+  simp only [delayed_refl_right,
+    trans_apply, refl_extend, path.coe_mk, function.comp_app, refl_apply],
+  split_ifs, swap, conv_rhs { rw ← γ.target },
+  all_goals { apply congr_arg γ, ext1, rw Q_right_zero_right },
+  exacts [if_neg h, if_pos h],
+end
+
+lemma delayed_refl_right_one (γ : path x y) : delayed_refl_right 1 γ = γ :=
+by { ext t, exact congr_arg γ (Q_right_one_right t) }
+
+
+def delayed_refl_left (θ : I) (γ : path x y) : path x y := (delayed_refl_right θ γ.symm).symm
+
+
+-- `[FAE]` This should probably be moved to `path_conneceted.lean`
+lemma continuous_symm {x y : X} : continuous (symm : (path x y) → (path y x)) :=
+begin
+ let f := @path.symm _ _ x y,
+--  let g := λ γ : path x y, γ ∘ σ,
+--  have : continuous f,
+--  have hg : continuous g := by continuity,
+ let g := λ p : path x y × I, (p.1 ∘ σ) p.2,
+ have hg : continuous g,
+ { exact continuous_fst.path_eval (continuous_symm.comp continuous_snd), },
+ let g' : C(path x y × I, X) := ⟨g, hg⟩,
+ let g'' := g'.curry,
+ convert g''.2 using 1,
+ sorry,
+end
```

Comment:
Like the proof of `continuous_trans`, this could simply be
```suggestion
lemma continuous_symm {x y : X} : continuous (symm : path x y → path y x) :=
continuous_uncurry_iff.mp $ path.symm_continuous_family _ (continuous_fst.path_eval continuous_snd)
```